Common/MathUtils/include/MathUtils/detail/trigonometric.h

```diff
@@ -107,14 +107,14 @@ inline void bringToPMPiGen(T& phi)
   phi = toPMPiGen<T>(phi);
 }
 
-GPUdi() void sincos(float ang, float& s, float& c)
-{
-  o2::gpu::GPUCommonMath::SinCos(ang, s, c);
-}
-
-GPUdi() void sincos(double ang, double& s, double& c)
+template <typename T>
+GPUdi() void sincos(T ang, T& s, T& c)
```

Comment:
Could you do this via template specialization and not via constexpr if std::is_same.
I am asking because I want to port also the other track model to the GPU eventually and std::is_same is not guaranteed to be available on the GPU in all cases :)
That should actually also fix the CI error.

Reply:
Will change it to your proposed solution - this is exactly what template specialization is for.